product_pricelist_show_product_ref/models/product_pricelist.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2020 Tecnativa - João Marques
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = "product.pricelist.item"
+
+    def _set_pricelist_item_name(self, product):
+        if product.product_id:
+            product.name = product.product_id.display_name
+
+    def _get_pricelist_item_name_price(self):
+        super(). _get_pricelist_item_name_price()
+        if len(self) > 1:
+            for product in self:
+                self._set_pricelist_item_name(product)
+        else:
+            self._set_pricelist_item_name(self)
```

Comment:
```suggestion
    def _get_pricelist_item_name_price(self):
        res = super(). _get_pricelist_item_name_price()
        for record in self.filtered("product_id"):
            record.name = product.product_id.display_name
        return res
```
I think this could be simplified to this, what do you think?
I think also that it's a good practice to get the value of super () ... and then return it even though it returns None (ping @pedrobaeza )

Reply:
In this case, the super() function doesn't return anything indeed, but we can add the call to increase resilience. 
Your other suggestion seems like a good idea. I've tested and it works, so I'm committing it